src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,40 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final List<EventListener> listeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executor = Executors.newFixedThreadPool(50);
 
+    public void registerListener(EventListener listener) {
+        listeners.add(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        for (EventListener e : listeners) {
+            try {
+                executor.execute(() -> e.onEvent(event));
+            } catch (Exception ex) {
+                throw new RuntimeException("Error notifying listener: " + e, ex);
+            }
+        }
     }
 
     public void shutdown() {
-
+        try {
+            if (executor.awaitTermination(1, TimeUnit.SECONDS)) {
+                executor.shutdownNow();
+            }
+        } catch (InterruptedException e) {
+            throw new RuntimeException("Error shutting down", e);
+        }
```

Comment:
this logic that you have inside this method awaits 1 second and then immidietly shuts down, returning the tasks that were still in queue, but we are ignoring them. I do not think that is what we want to do here, since our method is called `shutdown()` and not `shutdownNow()`. So let's just call `executor.shutdown()`, maybe in a try/catch block, but nothing else, to make it gracefully, as our method name suggests